app/components/autocomplete/emoji_suggestion/emoji_suggestion.js

```diff
@@ -92,7 +92,7 @@ export default class EmojiSuggestion extends Component {
             } else {
                 const initialEmojis = [...nextProps.emojis];
                 initialEmojis.splice(0, 300);
-                const data = initialEmojis.sort();
+                const data = initialEmojis.sort(compareEmojis);
```

Comment:
the comparison is being executed every time the component changes props (lets take out the fact that this `componentsWillReceiveProps` function will be removed as it has been deprecated), I think the better way to go about this is to apply the comparison to the selector here https://github.com/mattermost/mattermost-mobile/blob/master/app/components/autocomplete/emoji_suggestion/index.js#L26 then this line is not needed anymore